ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/Generators.hs

```diff
@@ -102,24 +101,11 @@ instance CanMock era => Arbitrary (Point (ShelleyBlock era)) where
 
 instance Era era => Arbitrary (TPraosState era) where
   arbitrary = do
-      steps     <- choose (0, 5)
-      startSlot <- frequency
+      lastSlot <- frequency
         [ (1, return Origin)
         , (5, NotOrigin . SlotNo <$> choose (0, 100))
         ]
-      initState <- TPraosState.empty startSlot <$> arbitrary
-      go steps startSlot initState
-    where
-      go :: Int
-         -> WithOrigin SlotNo
-         -> TPraosState era
-         -> Gen (TPraosState era)
-      go steps prevSlot st
-        | 0 <- steps = return st
-        | otherwise  = do
-          let slot = withOrigin (SlotNo 0) succ prevSlot
-          newPrtclState <- arbitrary
-          go (steps - 1) (NotOrigin slot) (TPraosState.append slot newPrtclState st)
+      TPraosState lastSlot <$> arbitrary
```

Comment:
I'm surprised that this generator can choose `lastSlot` independently of `tpraosStateChainDepState`.

Is it only used in properties testing low-level things like serialization eg?

Reply:
Exactly, it's only used for the serialisation roundtrip tests.